test/integration/compile-swagger-test.coffee

```diff
@@ -97,25 +97,44 @@ describe('compile() · Swagger', ->
   )
 
   describe('with multiple responses', ->
-    # Multiple responses is the closest we can get with Swagger to something
-    # like transaction examples ¯\_(ツ)_/¯
-
+    transactions = undefined
+    filename = 'apiDescription.json'
     detectTransactionExamples = sinon.spy(require('../../src/detect-transaction-examples'))
-    compilationResult = undefined
-    transaction = undefined
+
+    expectedStatusCodes = [200, 400, 500]
 
     beforeEach((done) ->
       stubs = {'./detect-transaction-examples': detectTransactionExamples}
 
-      compileFixture(fixtures.multipleResponses.swagger, {stubs}, (args...) ->
+      compileFixture(fixtures.multipleResponses.swagger, {filename, stubs}, (args...) ->
         [err, compilationResult] = args
-        transaction = compilationResult.transactions[0]
+        transactions = compilationResult?.transactions
         done(err)
       )
     )
 
     it('detection of transaction examples was not called', ->
       assert.isFalse(detectTransactionExamples.called)
     )
+    it('expected number of transactions was returned', ->
+      assert.equal(transactions.length, expectedStatusCodes.length)
+    )
+
+    for statusCode, i in expectedStatusCodes
+      do (statusCode, i) ->
+        context("origin of transaction ##{i + 1}", ->
+          it('uses URI as resource name', ->
+            assert.equal(transactions[i].origin.resourceName, '/honey')
+          )
+          it('uses method as action name', ->
+            assert.equal(transactions[i].origin.actionName, 'GET')
```

Comment:
Although I understand your point, I would dispute it with a fact that reviewer is unable to verify expected data unless he would:

1. Find where the fixtures is loaded
2. Find the source of the fixtures
3. Goes to the fixture file to verify the literal

Declaring variable with expected value near the place, where the fixture is required would omit first two steps and saves more brain cycles, IMHO :-)

With current setup, I have to convince myself to make a journey to verify the expected value, which could lead to skipping typos (`goat` vs `boat`; both can be legit) in false positive test.